osbuild/util/chroot.py

```diff
@@ -43,6 +51,11 @@ def __exit__(self, exc_type, exc_value, tracebk):
         if failed_umounts:
             print(f"Error unmounting paths from chroot: {failed_umounts}")
 
+        for d in self._bind_mounts[::-1]:
+            target_path = os.path.join(self.root, d.lstrip("/"))
+            if subprocess.run(["umount", "--lazy", target_path], check=False).returncode != 0:
+                print(f"Error unmounting paths from chroot: {d}")
```

Comment:
This should be moved up above the `if failed_umounts` block an put the failures in the `failed_umounts` list so the errors are consistent.